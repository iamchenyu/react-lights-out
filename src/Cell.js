import React from "react";
import "./Cell.css";

/** A single cell on the board.
 *
 * This has no state --- just two props:
 *
 * - flipCellsAroundMe: a function rec'd from the board which flips this
 *      cell and the cells around of it
 *
 * - isLit: boolean, is this cell lit?
 *
 * This handles clicks --- by calling flipCellsAroundMe
 *
 **/

// maybe not use testid because id should be unique ---> use role instead
function Cell({ flipCellsAroundMe, isLit }) {
  const classes = `Cell ${isLit ? "Cell-lit" : ""}`;
  return (
    <td
      className={classes}
      onClick={flipCellsAroundMe}
      data-testid="cell"
      role="button"
    />
  );
}

export default Cell;
