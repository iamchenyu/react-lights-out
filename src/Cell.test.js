import React from "react";
import { render } from "@testing-library/react";
import Cell from "./Cell";

// New version
// https://testing-library.com/docs/react-testing-library/api/#render
// append vs appendChild
// https://dev.to/ibn_abubakre/append-vs-appendchild-a4m

let container;

beforeEach(() => {
  const tr = document.createElement("tr");
  container = document.body.appendChild(tr);
});

it("renders without crashing", () => {
  render(<Cell />, { container });
});

it("matches the snapshot when lit", () => {
  const { asFragment } = render(<Cell isLit />, { container });
  expect(asFragment()).toMatchSnapshot();
});

it("matches the snapshot when unlit", () => {
  const { asFragment } = render(<Cell />, { container });
  expect(asFragment()).toMatchSnapshot();
});

// Old version
// it("renders without crashing", () => {
//   render(
//     <tr>
//       <Cell />
//     </tr>
//   );
// });

// it("matches the snapshot", () => {
//   const { asFragment } = render(
//     <tr>
//       <Cell />
//     </tr>
//   );
//   expect(asFragment()).toMatchSnapshot();
// });
