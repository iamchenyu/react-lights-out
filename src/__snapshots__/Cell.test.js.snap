// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot when lit 1`] = `<DocumentFragment />`;

exports[`matches the snapshot when unlit 1`] = `<DocumentFragment />`;
